geopandas/_vectorized.py

```diff
@@ -515,18 +515,31 @@ def geom_type(data):
         return _unary_op("geom_type", data, null_value=None)
 
 
-def area(data):
-    if compat.USE_PYGEOS:
-        return pygeos.area(data)
-    else:
-        return _unary_op("area", data, null_value=np.nan)
+def area(data, crs=None):
+    if crs and crs.is_geographic:
+        geod = crs.get_geod()
+        a_p = [geod.geometry_area_perimeter(geom) for geom in to_shapely(data)]
```

Comment:
Empty geometries are handled inconsistently here:  an empty polygon, line, or point raises an exception `pyproj.exceptions.GeodError: Invalid geometry provided.` whereas each returns an area of 0 in pygeos and `_unary_op`.

It may be necessary to wrap this in a try / catch and default the area to 0 for consistency with the others.

I didn't check what happens here in the self-intersecting geometry case, but it is specifically noted in the docs for `geometry_area_perimeter` so we should watch for consistency of results / exceptions there.

Having to go through shapely geometries here seems counter to moving toward pygeos internally.  It looks like it would be possible to use [`polygon_area_permiter`](https://pyproj4.github.io/pyproj/stable/api/geod.html#pyproj.Geod.polygon_area_perimeter) directly (is part of the public API).  This would allow extracting the x,y coordinate array for each geometry using something like `pygeos.get_coordinates(pg_geom)`.

This also appears to handle the case of empty geometries correctly:

```
>>> geod.polygon_area_perimeter([], [])
(0.0, 0.0)
```

Reply:
Good points, I'll explore options and add more tests.

Reply:
One thing we should be cautious about is unexpected interactions between geometry types and behavior in pyproj.  

For instance
> There’s no need to “close” the polygon by repeating the first vertex.
in the [docs](https://pyproj4.github.io/pyproj/stable/api/geod.html#pyproj.Geod.polygon_area_perimeter) makes me concerned that if we passed in a line, that it could get converted into a polygon and return an area - plus a longer perimeter for that last closing segment.

It seems easy to screen out points and lines from an area calculation, but GeometryCollection will be a problem.

Reply:
I will write our own function taking care of all geom types and loops through multi polygon using `polygon_area_perimeter` and `line_lengths` to ensure correct behaviour, but we may leave that for the 0.9. Not sure how quickly that will be done.